tests/form/test_validation.py

```diff
@@ -0,0 +1,83 @@
+import re
+
+import pytest
+from questionpy_common.elements import OptionsFormDefinition, StaticTextElement, FormSection, \
+    CheckboxElement, RepetitionElement, TextInputElement
+
+from questionpy.form import is_checked, equals
+from questionpy.form.validation import FormReferenceError, validate_form, FormError
+
+
+def test_should_not_raise_when_form_is_valid() -> None:
+    form = OptionsFormDefinition(
+        general=[CheckboxElement(name="chk1", hide_if=[is_checked("sect[chk2]")])],
+        sections=[FormSection(name="sect", header="",
+                              elements=[CheckboxElement(name="chk2", disable_if=[equals("..[chk1]", True)])])]
+    )
+
+    validate_form(form)
+
+
+def test_should_raise_FormReferenceError_when_node_doesnt_exist() -> None:
+    form_definition = OptionsFormDefinition(
+        general=[StaticTextElement(name="static", label="", text="", hide_if=[is_checked("sect[nonexistent]")])],
+        sections=[FormSection(name="sect", header="", elements=[])]
+    )
+
+    with pytest.raises(FormReferenceError) as exc_info:
+        validate_form(form_definition)
+
+    assert exc_info.value.node == "static"
+    assert exc_info.value.reference == "sect[nonexistent]"
+    assert exc_info.value.container_name == "sect"
+    assert exc_info.value.local_name == "nonexistent"
+
+
+def test_should_allow_reference_out_of_repetition() -> None:
+    form_definition = OptionsFormDefinition(general=[
+        CheckboxElement(name="chk"),
+        RepetitionElement(name="repetition", initial_elements=1, increment=1, elements=[
+            StaticTextElement(name="static", label="", text="", hide_if=[is_checked("..[chk]")]),
```

Comment:
Da es ja auch zwei Ebenen tief gehen kann e.g. `"..[..[chk]"`, ist es mit den eckigen Klammern vielleicht leichter.
Oder wie würde man das ein `parent` Element verweisen bei der Python Schreibweise?

Reply:
Guter Punkt (hehe). Das wäre dann vermutlich `...chk`, d.h. `removeprefix(".").split(".")`, und jeder leere string heißt "parent". Dann wäre
* `.chk` equivalent zu `chk`
* `..chk` würde im Parent suchen
* `...chk` würde im Parent des Parent suchen

Bis auf ersteres verhalten sich Python-Imports genauso. (`chk` wäre da natürlich absolut.) Ich halte absolute Referenzen in unserem Fall zwar für nicht besonders hilfreich, aber eigentlich gibt es auch keinen Grund, sie zu verbieten und wir könnten es komplett wie Python machen.

Oder wir definieren einen bestimmten string als "Parent-Selector", z.B. `^`: `^.^.chk`. Dann haben wir ein Smiley-Face. Vielleicht also nicht unbedingt `^`

Reply:
@MartinGauk hast du hier eine Meinung zu?

Reply:
Für mich ist das alles in Ordnung wie es ist. Denkbar wäre aber auch eine Schreibweise mit Schrägstrich `../../chk`

Reply:
Alles klar, dann würde ich bei den eckigen Klammern bleiben